spec/controllers/users_controller_spec.rb

```diff
@@ -100,6 +100,11 @@
               put :update, params: { id: user.to_param, user: { name: 'Trung Le', email: 'newmail@example.com' } }
             }.to change { ActionMailer::Base.deliveries.count }.by(1)
           end
+
+          it "allows updating the requested user" do
+            expect_any_instance_of(User).to receive(:update_attributes).with(ActionController::Parameters.new({ name: 'Trung Le' }).permit(:name))
```

Comment:
Personal taste, but I see the usage of `any_instance` as a code smell 😉 
In this case, you could have just gone with
```ruby
expect(user).to receive(:update_attributes)
```

However, I also don't see the point in mocking the call here. I think it would make more sense to actually test the outcome, so that updating actually works.
I'd suggest something in the direction of this:
```ruby
subject(:request) { put :update, params: params }
let(:params) { { id: user.id, user: { name: 'whatever' } }

it 'updates the user record' do
  expect { request }.to change(user, :name).from('before').to('whatever)
  expect(response).to redirect_to user
  expect(flash[:notice]).to eq 'Successfully updated your profile'
end
```

So more in the direction of testing that users can really update their profile.